### PR TITLE
Combat logic

### PR DESCRIPTION
**Pull request by Suedie**
Logic for damage
View properties of damage not implemented

### CODE REVIEW

File: core/src/model/LivingEntity.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+/**
+ * Parent class for all entities that have movement and health
+ */
+public abstract class LivingEntity extends Entity {
+    private float speed;
+    private boolean inMotion = false;
+    private float maxHealth;
+    private float currentHealth;
+    private float collisionDamage;
```
Comment (rojtekin) [inline core/src/model/LivingEntity.java]: Maybe keep it as only "damage" becasue the damage number should be the defualt one while things like weapons damage amplify/gets added onto this damage.

File: core/src/model/monsters/Monster.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package model.monsters;
+
+import model.LivingEntity;
+import model.Entity;
+import com.dongbat.jbump.World;
+
+public abstract class Monster extends LivingEntity {
+    //Bad practice? only works if static
+    public static final String DEFAULTFACTION = "monsters";
```
Comment (rojtekin) [inline core/src/model/monsters/Monster.java]: maybe better to use another way to determine the defualt faction?

**Review by rojtekin — APPROVED** [review]
LGTM

**Review by augnor — APPROVED** [review]
lgtm

File: core/src/model/monsters/Monster.java (hunk 2 above)
Comment (pkasinski) [reply core/src/model/monsters/Monster.java]: Have you considered creating an enum for the different factions?

**Pull request closed without merging**